chall12/vde-dios.c

```diff
@@ -0,0 +1,55 @@
+#include <string.h>
+
+void    swap_str(char *str, int start, int finish)
+{
+    char        aux;
+    int         i;
+
+    i = start + 1;
+    while (i < (finish - start + 1) / 2 + start)
+    {
+        aux = str[i];
+        str[i] = str[finish + start - i];
+        str[i] == ')' ? str[i] = '(' : (str[i] == '(' ? str[i] = ')' : str[i]);
+        aux == ')' ? aux = '(' : (aux == '(' ? aux = ')' : aux);
+        str[finish + start - i] = aux;
+        i++;
+    }
+    if ((finish - start + 1) % 2)
+        str[i] == ')' ? str[i] = '(' : (str[i] == '(' ? str[i] = ')' : str[i]);
+}
+
+char *ft_reverse_parenthesis(const char *str)
+{
+    int i, j, level;
+    char *aux = strdup(str);
```

Comment:
What if it fails?

Reply:
I do not really understand what  do you mean here. Do you mean if str is an invalid input? Do you suggest checking its validity before duplicating it?